content/documentation/pages/5-feature-guides/1-streams/9-function-composition.md

```diff
@@ -1,9 +1,331 @@
+
+```
+cd composed-http-transformer-rabbitmq
+./mvnw clean install
+```
+
+#### Registering the locally built application
+
+Now register `http-transformer` application by using the Data Flow Shell.
+
+**NOTE**
+
+> For the below app register `--uri` option, replace the directory name and path of the artifact with the value appropriate to your system.
+
+```
+app register --name http-transformer --type source --uri file:///>YOUR-SOURCE-CODE>/target/composed-http-transformer-[kafka/rabbitmq]-0.0.1-SNAPSHOT.jar
+```
+
+#### Registering the readily available application
+
+The maven/docker artifacts of the `http-transformer` application are readily available in both the `Kafka` and `RabbitMQ` binders.
+
+Maven artifact with Kafka binder:
+
+```
+app register --name http-transformer --type source --uri maven://io.spring.dataflow.sample:composed-http-transformer-kafka:0.0.1-SNAPSHOT
+```
+
+Maven artifact with RabbitMQ binder:
+
+```
+app register --name http-transformer --type source --uri maven://io.spring.dataflow.sample:composed-http-transformer-rabbitmq:0.0.1-SNAPSHOT
+```
+
+Docker artifact with Kafka binder:
+
+```
+app register --name http-transformer --type source --uri docker://springcloudstream/composed-http-transformer-kafka:0.0.1-SNAPSHOT
+```
+
+Docker artifact with RabbitMQ binder:
+
+```
+app register --name http-transformer --type source --uri docker://springcloudstream/composed-http-transformer-rabbitmq:0.0.1-SNAPSHOT
+```
+
+#### Deploying the stream
+
+Let's create the stream without the explicit `transform` application from the previous example. Instead, use the `http-transform` that has the functional beans named `upper` and `concat` defined.
+
+```
+stream create helloComposed --definition "http-transformer --server.port=9001 | log"
+```
+
+To compose a function bean to the application, we need to pass the `spring.cloud.stream.function.definition` property to define the composed function DSL (defined as in Spring Cloud Function).
+
+In this case, it is:
+
+```
+stream deploy helloComposed --properties "app.http-transformer.spring.cloud.stream.function.definition=upper|concat"
+
+```
+
+The above deployment composes the `upper` and `concat` function beans into the `http` source application.
+
+Then we can send the payload to `http` application:
+
+```
+http post --data "friend" --target "http://localhost:9001"
+```
+
+Then you can see the output in the `log` application as,
+
+```
+[helloComposed-1] log-sink                                 : Hello FRIEND
+
+```
+
+**NOTE**
+
+Please note that function composition support is not applicable for the out-of-the-box Spring Cloud Stream `Processor` applications, since there is ambiguity in whether the function needs to be applied before or after the existing processor’s application logic.
+
+However, you can create your own processor applications that use functional composition with standard java.util.Function APIs, as the following example shows:
```

Comment:
How is it related to the previous example? Is this a new App? If yes, it'd be good to clarify it before getting into the details.

Reply:
It is neither related to the previous example nor a new app. It just shows how one can use compose functional beans on custom `processor` applications. I will add this to a separate section to clarify it better.